src/java.base/share/classes/jdk/internal/foreign/abi/Binding.java

```diff
@@ -348,15 +348,15 @@ static Allocate allocate(MemoryLayout layout) {
     }
 
     static BoxAddress boxAddressRaw(long size) {
-        return new BoxAddress(size, false);
+        return new BoxAddress(size, 1, false);
```

Comment:
Now that we have the full layout, we might want to add an overload of `boxAddressRaw` that takes a pointee layout, and takes the alignment into account as well. (or just change this method to accept an alignment as well). Most of the call sites of this method are passing the result of `Utils.pointeeSize(layout)` (where layout is the address layout).

Also, the main difference with `boxAddress` seems to be the lack of scope being used. So, maybe it would be better to name this method `boxAddressNoScope`.

Reply:
I have fixed this in the latest iteration (but I left the old name) - as I saw test failures because of this